<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
      
        <link rel="next" href="cliargs/">
      
      
      <link rel="icon" href="assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.15">
    
    
      
        <title>vcfstats</title>
      
    
    
      <link rel="stylesheet" href="assets/stylesheets/main.342714a4.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Ubuntu:300,300i,400,400i,700,700i%7CUbuntu+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Ubuntu";--md-code-font:"Ubuntu Mono"}</style>
      
    
    
      <link rel="stylesheet" href="style.css">
    
    <script>__md_scope=new URL(".",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#vcfstats-powerful-statistics-for-vcf-files" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="." title="vcfstats" class="md-header__button md-logo" aria-label="vcfstats" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            vcfstats
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Introduction
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/pwwang/vcfstats" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    pwwang/vcfstats
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="." title="vcfstats" class="md-nav__button md-logo" aria-label="vcfstats" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    vcfstats
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/pwwang/vcfstats" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    pwwang/vcfstats
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="." class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#vcfstats-powerful-statistics-for-vcf-files" class="md-nav__link">
    <span class="md-ellipsis">
      vcfstats - powerful statistics for VCF files
    </span>
  </a>
  
    <nav class="md-nav" aria-label="vcfstats - powerful statistics for VCF files">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#motivation" class="md-nav__link">
    <span class="md-ellipsis">
      Motivation
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#installation" class="md-nav__link">
    <span class="md-ellipsis">
      Installation
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gallery" class="md-nav__link">
    <span class="md-ellipsis">
      Gallery
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Gallery">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#number-of-variants-on-each-chromosome" class="md-nav__link">
    <span class="md-ellipsis">
      Number of variants on each chromosome
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Number of variants on each chromosome">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#changing-labels-and-ticks" class="md-nav__link">
    <span class="md-ellipsis">
      Changing labels and ticks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#number-of-variants-on-first-5-chromosome" class="md-nav__link">
    <span class="md-ellipsis">
      Number of variants on first 5 chromosome
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#number-of-substitutions-of-snps" class="md-nav__link">
    <span class="md-ellipsis">
      Number of substitutions of SNPs
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Number of substitutions of SNPs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#only-with-snps-pass-all-filters" class="md-nav__link">
    <span class="md-ellipsis">
      Only with SNPs PASS all filters
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#alternative-allele-frequency-on-each-chromosome" class="md-nav__link">
    <span class="md-ellipsis">
      Alternative allele frequency on each chromosome
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Alternative allele frequency on each chromosome">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#using-boxplot" class="md-nav__link">
    <span class="md-ellipsis">
      Using boxplot
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#using-density-plothistogram-to-investigate-the-distribution" class="md-nav__link">
    <span class="md-ellipsis">
      Using density plot/histogram to investigate the distribution:
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#overall-distribution-of-allele-frequency" class="md-nav__link">
    <span class="md-ellipsis">
      Overall distribution of allele frequency
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Overall distribution of allele frequency">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#excluding-some-lowhigh-frequency-variants" class="md-nav__link">
    <span class="md-ellipsis">
      Excluding some low/high frequency variants
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#counting-types-of-variants-on-each-chromosome" class="md-nav__link">
    <span class="md-ellipsis">
      Counting types of variants on each chromosome
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Counting types of variants on each chromosome">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#using-bar-chart-if-there-is-only-one-chromosome" class="md-nav__link">
    <span class="md-ellipsis">
      Using bar chart if there is only one chromosome
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#counting-variant-types-on-whole-genome" class="md-nav__link">
    <span class="md-ellipsis">
      Counting variant types on whole genome
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#counting-type-of-mutant-genotypes-het-hom_alt-for-sample-1-on-each-chromosome" class="md-nav__link">
    <span class="md-ellipsis">
      Counting type of mutant genotypes (HET, HOM_ALT) for sample 1 on each chromosome
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#exploration-of-meangenotype-quality-and-meandepth-on-each-chromosome-for-sample-1" class="md-nav__link">
    <span class="md-ellipsis">
      Exploration of mean(genotype quality) and mean(depth) on each chromosome for sample 1
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#exploration-of-depths-for-sample-12" class="md-nav__link">
    <span class="md-ellipsis">
      Exploration of depths for sample 1,2
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="cliargs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Command-line arguments
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="configfile/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Configuration file
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="formulas/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Formulas
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="macros/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Macros
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="CHANGELOG/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Change log
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#vcfstats-powerful-statistics-for-vcf-files" class="md-nav__link">
    <span class="md-ellipsis">
      vcfstats - powerful statistics for VCF files
    </span>
  </a>
  
    <nav class="md-nav" aria-label="vcfstats - powerful statistics for VCF files">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#motivation" class="md-nav__link">
    <span class="md-ellipsis">
      Motivation
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#installation" class="md-nav__link">
    <span class="md-ellipsis">
      Installation
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gallery" class="md-nav__link">
    <span class="md-ellipsis">
      Gallery
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Gallery">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#number-of-variants-on-each-chromosome" class="md-nav__link">
    <span class="md-ellipsis">
      Number of variants on each chromosome
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Number of variants on each chromosome">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#changing-labels-and-ticks" class="md-nav__link">
    <span class="md-ellipsis">
      Changing labels and ticks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#number-of-variants-on-first-5-chromosome" class="md-nav__link">
    <span class="md-ellipsis">
      Number of variants on first 5 chromosome
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#number-of-substitutions-of-snps" class="md-nav__link">
    <span class="md-ellipsis">
      Number of substitutions of SNPs
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Number of substitutions of SNPs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#only-with-snps-pass-all-filters" class="md-nav__link">
    <span class="md-ellipsis">
      Only with SNPs PASS all filters
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#alternative-allele-frequency-on-each-chromosome" class="md-nav__link">
    <span class="md-ellipsis">
      Alternative allele frequency on each chromosome
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Alternative allele frequency on each chromosome">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#using-boxplot" class="md-nav__link">
    <span class="md-ellipsis">
      Using boxplot
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#using-density-plothistogram-to-investigate-the-distribution" class="md-nav__link">
    <span class="md-ellipsis">
      Using density plot/histogram to investigate the distribution:
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#overall-distribution-of-allele-frequency" class="md-nav__link">
    <span class="md-ellipsis">
      Overall distribution of allele frequency
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Overall distribution of allele frequency">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#excluding-some-lowhigh-frequency-variants" class="md-nav__link">
    <span class="md-ellipsis">
      Excluding some low/high frequency variants
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#counting-types-of-variants-on-each-chromosome" class="md-nav__link">
    <span class="md-ellipsis">
      Counting types of variants on each chromosome
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Counting types of variants on each chromosome">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#using-bar-chart-if-there-is-only-one-chromosome" class="md-nav__link">
    <span class="md-ellipsis">
      Using bar chart if there is only one chromosome
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#counting-variant-types-on-whole-genome" class="md-nav__link">
    <span class="md-ellipsis">
      Counting variant types on whole genome
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#counting-type-of-mutant-genotypes-het-hom_alt-for-sample-1-on-each-chromosome" class="md-nav__link">
    <span class="md-ellipsis">
      Counting type of mutant genotypes (HET, HOM_ALT) for sample 1 on each chromosome
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#exploration-of-meangenotype-quality-and-meandepth-on-each-chromosome-for-sample-1" class="md-nav__link">
    <span class="md-ellipsis">
      Exploration of mean(genotype quality) and mean(depth) on each chromosome for sample 1
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#exploration-of-depths-for-sample-12" class="md-nav__link">
    <span class="md-ellipsis">
      Exploration of depths for sample 1,2
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


  <h1>Introduction</h1>

<h2 id="vcfstats-powerful-statistics-for-vcf-files">vcfstats - powerful statistics for VCF files</h2>
<p><a href="https://pypi.org/project/vcfstats/"><img alt="Pypi" src="https://img.shields.io/pypi/v/vcfstats?style=flat-square" /></a> <a href="https://github.com/pwwang/vcfstats"><img alt="Github" src="https://img.shields.io/github/v/tag/pwwang/vcfstats?style=flat-square" /></a> <a href="https://pypi.org/project/vcfstats/"><img alt="PythonVers" src="https://img.shields.io/pypi/pyversions/vcfstats?style=flat-square" /></a> <a href="https://pwwang.github.io/vcfstats/"><img alt="docs" src="https://img.shields.io/github/actions/workflow/status/pwwang/vcfstats/docs.yml?label=docs&amp;style=flat-square" /></a> <img alt="github action" src="https://img.shields.io/github/actions/workflow/status/pwwang/vcfstats/build.yml?style=flat-square" /> [<img alt="Codacy" src="https://img.shields.io/codacy/grade/c8c8bfa8c5e9443bbf268a0a7c6f206d?style=flat-square" />]<a href="https://app.codacy.com/gh/pwwang/vcfstats/">10</a> [<img alt="Codacy coverage" src="https://img.shields.io/codacy/coverage/c8c8bfa8c5e9443bbf268a0a7c6f206d?style=flat-square" />]<a href="https://app.codacy.com/gh/pwwang/vcfstats/">10</a></p>
<p><a href="https://pwwang.github.io/vcfstats/">Documentation</a> | <a href="https://pwwang.github.io/vcfstats/CHANGELOG/">CHANGELOG</a></p>
<h3 id="motivation">Motivation</h3>
<p>There are a couple of tools that can plot some statistics of VCF files, including <a href="https://samtools.github.io/bcftools/bcftools.html#stats"><code>bcftools</code></a> and <a href="http://lindenb.github.io/jvarkit/VcfStatsJfx.html"><code>jvarkit</code></a>. However, none of them could:</p>
<ol>
<li>plot specific metrics</li>
<li>customize the plots</li>
<li>focus on variants with certain filters</li>
</ol>
<p>R package <a href="https://knausb.github.io/vcfR_documentation/visualization_1.html"><code>vcfR</code></a> can do some of the above. However, it has to load entire VCF into memory, which is not friendly to large VCF files.</p>
<h3 id="installation">Installation</h3>
<div class="highlight"><pre><span></span><code>pip<span class="w"> </span>install<span class="w"> </span>-U<span class="w"> </span>vcfstats
</code></pre></div>
<p>Or run with docker:</p>
<div class="highlight"><pre><span></span><code>docker<span class="w"> </span>run<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-w<span class="w"> </span>/vcfstats/workdir<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-v<span class="w"> </span><span class="k">$(</span><span class="nb">pwd</span><span class="k">)</span>:/vcfstats/workdir<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--rm<span class="w"> </span>justold/vcfstats:latest<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>vcfstats<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--vcf<span class="w"> </span>myfile.vcf<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-o<span class="w"> </span>outputs<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--formula<span class="w"> </span><span class="s1">&#39;COUNT(1) ~ CONTIG&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--title<span class="w"> </span><span class="s1">&#39;Number of variants on each chromosome&#39;</span>
</code></pre></div>
<h3 id="gallery">Gallery</h3>
<h4 id="number-of-variants-on-each-chromosome">Number of variants on each chromosome</h4>
<div class="highlight"><pre><span></span><code>vcfstats<span class="w"> </span>--vcf<span class="w"> </span>examples/sample.vcf<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--outdir<span class="w"> </span>examples/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--formula<span class="w"> </span><span class="s1">&#39;COUNT(1) ~ CONTIG&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--title<span class="w"> </span><span class="s1">&#39;Number of variants on each chromosome&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--config<span class="w"> </span>examples/config.toml
</code></pre></div>
<p><img alt="Number of variants on each chromosome" src="examples/number-of-variants-on-each-chromosome.col.png" /></p>
<h5 id="changing-labels-and-ticks">Changing labels and ticks</h5>
<p><code>vcfstats</code> uses <a href="https://plotnine.readthedocs.io/en/stable/"><code>plotnine</code></a> for plotting, read more about it on how to specify <code>--ggs</code> to modify the plots.</p>
<div class="highlight"><pre><span></span><code>vcfstats<span class="w"> </span>--vcf<span class="w"> </span>examples/sample.vcf<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--outdir<span class="w"> </span>examples/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--formula<span class="w"> </span><span class="s1">&#39;COUNT(1) ~ CONTIG&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--title<span class="w"> </span><span class="s1">&#39;Number of variants on each chromosome (modified)&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--config<span class="w"> </span>examples/config.toml<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--ggs<span class="w"> </span><span class="s1">&#39;scale_x_discrete(name =&quot;Chromosome&quot;, \</span>
<span class="s1">        limits=[&quot;1&quot;,&quot;2&quot;,&quot;3&quot;,&quot;4&quot;,&quot;5&quot;,&quot;6&quot;,&quot;7&quot;,&quot;8&quot;,&quot;9&quot;,&quot;10&quot;,&quot;X&quot;]); \</span>
<span class="s1">        ylab(&quot;# Variants&quot;)&#39;</span>
</code></pre></div>
<p><img alt="Number of variants on each chromosome (modified)" src="examples/number-of-variants-on-each-chromosome-modified.col.png" /></p>
<h5 id="number-of-variants-on-first-5-chromosome">Number of variants on first 5 chromosome</h5>
<div class="highlight"><pre><span></span><code>vcfstats<span class="w"> </span>--vcf<span class="w"> </span>examples/sample.vcf<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--outdir<span class="w"> </span>examples/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--formula<span class="w"> </span><span class="s1">&#39;COUNT(1) ~ CONTIG[1,2,3,4,5]&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--title<span class="w"> </span><span class="s1">&#39;Number of variants on each chromosome (first 5)&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--config<span class="w"> </span>examples/config.toml
<span class="c1"># or</span>
vcfstats<span class="w"> </span>--vcf<span class="w"> </span>examples/sample.vcf<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--outdir<span class="w"> </span>examples/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--formula<span class="w"> </span><span class="s1">&#39;COUNT(1) ~ CONTIG[1-5]&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--title<span class="w"> </span><span class="s1">&#39;Number of variants on each chromosome (first 5)&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--config<span class="w"> </span>examples/config.toml
<span class="c1"># or</span>
<span class="c1"># require vcf file to be tabix-indexed.</span>
vcfstats<span class="w"> </span>--vcf<span class="w"> </span>examples/sample.vcf<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--outdir<span class="w"> </span>examples/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--formula<span class="w"> </span><span class="s1">&#39;COUNT(1) ~ CONTIG&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--title<span class="w"> </span><span class="s1">&#39;Number of variants on each chromosome (first 5)&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--config<span class="w"> </span>examples/config.toml<span class="w"> </span>-r<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">3</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="m">5</span>
</code></pre></div>
<p><img alt="Number of variants on each chromosome (first 5)" src="examples/number-of-variants-on-each-chromosome-first-5.col.png" /></p>
<h4 id="number-of-substitutions-of-snps">Number of substitutions of SNPs</h4>
<div class="highlight"><pre><span></span><code>vcfstats<span class="w"> </span>--vcf<span class="w"> </span>examples/sample.vcf<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--outdir<span class="w"> </span>examples/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--formula<span class="w"> </span><span class="s1">&#39;COUNT(1, VARTYPE[snp]) ~ SUBST[A&gt;T,A&gt;G,A&gt;C,T&gt;A,T&gt;G,T&gt;C,G&gt;A,G&gt;T,G&gt;C,C&gt;A,C&gt;T,C&gt;G]&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--title<span class="w"> </span><span class="s1">&#39;Number of substitutions of SNPs&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--config<span class="w"> </span>examples/config.toml
</code></pre></div>
<p><img alt="Number of substitutions of SNPs" src="examples/number-of-substitutions-of-snps.col.png" /></p>
<h5 id="only-with-snps-pass-all-filters">Only with SNPs PASS all filters</h5>
<div class="highlight"><pre><span></span><code>vcfstats<span class="w"> </span>--vcf<span class="w"> </span>examples/sample.vcf<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--outdir<span class="w"> </span>examples/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--formula<span class="w"> </span><span class="s1">&#39;COUNT(1, VARTYPE[snp]) ~ SUBST[A&gt;T,A&gt;G,A&gt;C,T&gt;A,T&gt;G,T&gt;C,G&gt;A,G&gt;T,G&gt;C,C&gt;A,C&gt;T,C&gt;G]&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--title<span class="w"> </span><span class="s1">&#39;Number of substitutions of SNPs (passed)&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--config<span class="w"> </span>examples/config.toml<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--passed
</code></pre></div>
<p><img alt="Number of substitutions of SNPs (passed)" src="examples/number-of-substitutions-of-snps-passed.col.png" /></p>
<h4 id="alternative-allele-frequency-on-each-chromosome">Alternative allele frequency on each chromosome</h4>
<div class="highlight"><pre><span></span><code><span class="c1"># using a dark theme</span>
vcfstats<span class="w"> </span>--vcf<span class="w"> </span>examples/sample.vcf<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--outdir<span class="w"> </span>examples/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--formula<span class="w"> </span><span class="s1">&#39;AAF ~ CONTIG&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--title<span class="w"> </span><span class="s1">&#39;Allele frequency on each chromosome&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--config<span class="w"> </span>examples/config.toml<span class="w"> </span>--ggs<span class="w"> </span><span class="s1">&#39;theme_dark()&#39;</span>
</code></pre></div>
<p><img alt="Allele frequency on each chromosome" src="examples/allele-frequency-on-each-chromosome.violin.png" /></p>
<h5 id="using-boxplot">Using boxplot</h5>
<div class="highlight"><pre><span></span><code>vcfstats<span class="w"> </span>--vcf<span class="w"> </span>examples/sample.vcf<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--outdir<span class="w"> </span>examples/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--formula<span class="w"> </span><span class="s1">&#39;AAF ~ CONTIG&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--title<span class="w"> </span><span class="s1">&#39;Allele frequency on each chromosome (boxplot)&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--config<span class="w"> </span>examples/config.toml<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--figtype<span class="w"> </span>boxplot
</code></pre></div>
<p><img alt="Allele frequency on each chromosome" src="examples/allele-frequency-on-each-chromosome-boxplot.boxplot.png" /></p>
<h5 id="using-density-plothistogram-to-investigate-the-distribution">Using density plot/histogram to investigate the distribution:</h5>
<p>You can plot the distribution, using density plot or histogram</p>
<div class="highlight"><pre><span></span><code>vcfstats<span class="w"> </span>--vcf<span class="w"> </span>examples/sample.vcf<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--outdir<span class="w"> </span>examples/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--formula<span class="w"> </span><span class="s1">&#39;AAF ~ CONTIG[1,2]&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--title<span class="w"> </span><span class="s1">&#39;Allele frequency on chromosome 1,2&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--config<span class="w"> </span>examples/config.toml<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--figtype<span class="w"> </span>density
</code></pre></div>
<p><img alt="Allele frequency on chromosome 1,2" src="examples/allele-frequency-on-chromosome-12.density.png" /></p>
<h4 id="overall-distribution-of-allele-frequency">Overall distribution of allele frequency</h4>
<div class="highlight"><pre><span></span><code>vcfstats<span class="w"> </span>--vcf<span class="w"> </span>examples/sample.vcf<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--outdir<span class="w"> </span>examples/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--formula<span class="w"> </span><span class="s1">&#39;AAF ~ 1&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--title<span class="w"> </span><span class="s1">&#39;Overall allele frequency distribution&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--config<span class="w"> </span>examples/config.toml
</code></pre></div>
<p><img alt="Overall allele frequency distribution" src="examples/overall-allele-frequency-distribution.histogram.png" /></p>
<h5 id="excluding-some-lowhigh-frequency-variants">Excluding some low/high frequency variants</h5>
<div class="highlight"><pre><span></span><code>vcfstats<span class="w"> </span>--vcf<span class="w"> </span>examples/sample.vcf<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--outdir<span class="w"> </span>examples/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--formula<span class="w"> </span><span class="s1">&#39;AAF[0.05, 0.95] ~ 1&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--title<span class="w"> </span><span class="s1">&#39;Overall allele frequency distribution (0.05-0.95)&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--config<span class="w"> </span>examples/config.toml
</code></pre></div>
<p><img alt="Overall allele frequency distribution" src="examples/overall-allele-frequency-distribution-0-05-0-95.histogram.png" /></p>
<h4 id="counting-types-of-variants-on-each-chromosome">Counting types of variants on each chromosome</h4>
<div class="highlight"><pre><span></span><code>vcfstats<span class="w"> </span>--vcf<span class="w"> </span>examples/sample.vcf<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--outdir<span class="w"> </span>examples/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--formula<span class="w"> </span><span class="s1">&#39;COUNT(1, group=VARTYPE) ~ CHROM&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span><span class="c1"># or simply</span>
<span class="w">    </span><span class="c1"># --formula &#39;VARTYPE ~ CHROM&#39; \</span>
<span class="w">    </span>--title<span class="w"> </span><span class="s1">&#39;Types of variants on each chromosome&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--config<span class="w"> </span>examples/config.toml
</code></pre></div>
<p><img alt="Types of variants on each chromosome" src="examples/types-of-variants-on-each-chromosome.col.png" /></p>
<h5 id="using-bar-chart-if-there-is-only-one-chromosome">Using bar chart if there is only one chromosome</h5>
<div class="highlight"><pre><span></span><code>vcfstats<span class="w"> </span>--vcf<span class="w"> </span>examples/sample.vcf<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--outdir<span class="w"> </span>examples/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--formula<span class="w"> </span><span class="s1">&#39;COUNT(1, group=VARTYPE) ~ CHROM[1]&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span><span class="c1"># or simply</span>
<span class="w">    </span><span class="c1"># --formula &#39;VARTYPE ~ CHROM[1]&#39; \</span>
<span class="w">    </span>--title<span class="w"> </span><span class="s1">&#39;Types of variants on chromosome 1&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--config<span class="w"> </span>examples/config.toml<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--figtype<span class="w"> </span>pie
</code></pre></div>
<p><img alt="Types of variants on chromosome 1" src="examples/types-of-variants-on-chromosome-1.pie.png" /></p>
<h5 id="counting-variant-types-on-whole-genome">Counting variant types on whole genome</h5>
<div class="highlight"><pre><span></span><code>vcfstats<span class="w"> </span>--vcf<span class="w"> </span>examples/sample.vcf<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--outdir<span class="w"> </span>examples/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span><span class="c1"># or simply</span>
<span class="w">    </span><span class="c1"># --formula &#39;VARTYPE ~ 1&#39; \</span>
<span class="w">    </span>--formula<span class="w"> </span><span class="s1">&#39;COUNT(1, group=VARTYPE) ~ 1&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--title<span class="w"> </span><span class="s1">&#39;Types of variants on whole genome&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--config<span class="w"> </span>examples/config.toml
</code></pre></div>
<p><img alt="Types of variants on whole genome" src="examples/types-of-variants-on-whole-genome.col.png" /></p>
<h4 id="counting-type-of-mutant-genotypes-het-hom_alt-for-sample-1-on-each-chromosome">Counting type of mutant genotypes (HET, HOM_ALT) for sample 1 on each chromosome</h4>
<div class="highlight"><pre><span></span><code>vcfstats<span class="w"> </span>--vcf<span class="w"> </span>examples/sample.vcf<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--outdir<span class="w"> </span>examples/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span><span class="c1"># or simply</span>
<span class="w">    </span><span class="c1"># --formula &#39;GTTYPEs[HET,HOM_ALT]{0} ~ CHROM&#39; \</span>
<span class="w">    </span>--formula<span class="w"> </span><span class="s1">&#39;COUNT(1, group=GTTYPEs[HET,HOM_ALT]{0}) ~ CHROM&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--title<span class="w"> </span><span class="s1">&#39;Mutant genotypes on each chromosome (sample 1)&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--config<span class="w"> </span>examples/config.toml
</code></pre></div>
<p><img alt="Mutant genotypes on each chromosome" src="examples/mutant-genotypes-on-each-chromosome-sample-1.col.png" /></p>
<h4 id="exploration-of-meangenotype-quality-and-meandepth-on-each-chromosome-for-sample-1">Exploration of mean(genotype quality) and mean(depth) on each chromosome for sample 1</h4>
<div class="highlight"><pre><span></span><code>vcfstats<span class="w"> </span>--vcf<span class="w"> </span>examples/sample.vcf<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--outdir<span class="w"> </span>examples/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--formula<span class="w"> </span><span class="s1">&#39;MEAN(GQs{0}) ~ MEAN(DEPTHs{0}, group=CHROM)&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--title<span class="w"> </span><span class="s1">&#39;GQ vs depth (sample 1)&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--config<span class="w"> </span>examples/config.toml
</code></pre></div>
<p><img alt="GQ vs depth (sample 1)" src="examples/gq-vs-depth-sample-1.scatter.png" /></p>
<h4 id="exploration-of-depths-for-sample-12">Exploration of depths for sample 1,2</h4>
<div class="highlight"><pre><span></span><code>vcfstats<span class="w"> </span>--vcf<span class="w"> </span>examples/sample.vcf<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--outdir<span class="w"> </span>examples/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--formula<span class="w"> </span><span class="s1">&#39;DEPTHs{0} ~ DEPTHs{1}&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--title<span class="w"> </span><span class="s1">&#39;Depths between sample 1 and 2&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--config<span class="w"> </span>examples/config.toml
</code></pre></div>
<p><img alt="Depths between sample 1 and 2" src="examples/depths-between-sample-1-and-2.scatter.png" /></p>
<p>See more examples:</p>
<p><a href="https://github.com/pwwang/vcfstats/issues/15#issuecomment-1029367903">https://github.com/pwwang/vcfstats/issues/15#issuecomment-1029367903</a></p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": ".", "features": [], "search": "assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="assets/javascripts/bundle.56ea9cef.min.js"></script>
      
    
  </body>
</html>